cmd/opentelemetry-collector/app/defaults/default_config_test.go

```diff
@@ -162,6 +165,103 @@ func TestDefaultCollectorConfig(t *testing.T) {
 	}
 }
 
+func TestCreateCollectorReceivers(t *testing.T) {
+	tests := []struct {
+		name           string
+		args           []string
+		zipkinHostPort string
+		receivers      configmodels.Receivers
+	}{
+		{
+			name:           "defaultWithoutZipkin",
+			args:           []string{},
+			zipkinHostPort: ":0",
+			receivers: configmodels.Receivers{
+				"jaeger": &jaegerreceiver.Config{
+					TypeVal: "jaeger",
+					NameVal: "jaeger",
+					Protocols: map[string]*receiver.SecureReceiverSettings{
+						"grpc": {
+							ReceiverSettings: configmodels.ReceiverSettings{
+								Endpoint: gRPCEndpoint,
+							},
+						},
+						"thrift_http": {
+							ReceiverSettings: configmodels.ReceiverSettings{
+								Endpoint: httpThriftBinaryEndpoint,
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "configurationViaFlags",
+			args: []string{
+				"--collector.grpc-server.host-port=host:11",
+				"--collector.grpc.tls.cert=cacert.crt",
+				"--collector.grpc.tls.key=keycert.crt",
+				"--collector.http-server.host-port=host2:22",
+				"--reporter.grpc.host-port=coll:33",
```

Comment:
Might be missing something, but why would these `reporter` flags be defined on the collector? They are normally only supported to an agent.
I'm guessing its for when the collector maybe forwarding data to another collector using those reporter (or exporter) details - but think it may need some explanation of why these exporter connection details are being used for the receiver/remote sampling endpoint?

Reply:
Collector and agent uses OTEL Jaeger receiver. Jaeger receiver exposes HTTP sampling endpoint. It can be configured to get the strategies from a remote server via gRPC. Legacy collector was using the reporter (and its flags) to get the data from the collector. 

I can update the test and move the flags to agent test, but it really does not matter in the end.